library/ssl_cli.c

```diff
@@ -2383,6 +2383,10 @@ static int ssl_parse_server_ecdh_params_psa( mbedtls_ssl_context *ssl,
     tls_id <<= 8;
     tls_id |= *(*p)++;
 
+    /* Check it's a curve we offered */
+    if( mbedtls_ssl_check_curve_tls_id( ssl, tls_id ) != 0 )
+        return( MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE );
```

Comment:
We are altering behaviour from incorrect to correct, and the test will fail only if the server is misbehaving. OTOH, a debug message doesn't hurt, since they can easily be compiled out, so I'll add one.